sw/device/tests/crypto/cryptotest/firmware/kmac_sca.c

```diff
@@ -466,26 +495,56 @@ status_t handle_kmac_sca_set_key(ujson_t *uj) {
 /**
  * Absorbs a message using KMAC128 without a customization string.
  *
+ * Two modes are supported: FPGA and non-FPGA mode.
+ * The FPGA mode can be used to improve KMAC SCA. Here, features only available
+ * on the FPGA bitstream are used (i.e., SecKmacCmdDelay and
+ * SecKmacIdleAcceptSwMsg). The non-FPGA mode represents the chip where these
+ * features are disabled.
+ *
+ * In the FPGA mode, this function performs:
+ * - Configure mode/key/...
+ * - Write data to the message FIFO.
+ * - Send START and PROCESS commands, put Ibex into sleep.
+ *
+ * In the non-FPGA mode, this function performs:
+ * - Configure mode/key/...
+ * - Send START command, put Ibex into sleep.
+ * - Write message FIFO.
+ * - Send PROCESS command, put Ibex into sleep.
+ *
  * @param msg Message.
  * @param msg_len Message length.
  */
 static kmac_sca_error_t sha3_ujson_absorb(const uint8_t *msg, size_t msg_len) {
-  // Start a new message and write data to message FIFO.
+  // Start a new message.
   if (kmac_msg_start(kDifKmacModeKmacLen128, kDigestLength, &kmac_key, NULL) !=
       kmacScaOk) {
     return kmacScaAborted;
   }
 
+  if (fpga_mode == false) {
+    // Start command. On the chip, we need to first issue a START command
+    // before writing to the message FIFO.
+    sca_call_and_sleep(kmac_start_cmd, kIbexSha3SleepCycles);
```

Comment:
We may want to add to sets of `kIbexSha3SleepCycles` sleep constants for non-FPGA and  FPGA mode. The FPGA mode one shouldn't change with this PR. The non-FPGA one is probably split into two constants (to be verified experimentally):
1. The time required to load and hash the prefix + the time required to load and hash the key. Loading and padding is always 1600 bits / 64 bits/cycle = 25 cycles, processing is always 24 rounds * 4 cycles / round = 96 cycles. So this should be something like 300 to provide some ample cycles
2. The time required to load and hash the message, i.e., again 121 cycles, maybe take again 150 to be on the safe side.

Reply:
Thanks for the cycles! I've changed the sleep cycle parameters for the non-FPGA mode and I will evaluate this soon.

Reply:
I've done some experiments here:
https://github.com/lowRISC/opentitan/pull/21093
If everything looks fine, could you please merge this?